variants/frontend-stimulus/template.rb

```diff
@@ -0,0 +1,70 @@
+source_paths.unshift(File.dirname(__FILE__))
+
+puts "Setting up stimulus.js"
+
+yarn_add_dependencies %w[
+  @hotwired/stimulus
+  @hotwired/stimulus-webpack-helpers
+]
+
+if TEMPLATE_CONFIG.use_typescript?
+  copy_file "variants/frontend-stimulus/hello_controller.ts", "app/frontend/javascript/controllers/hello_controller.ts"
+else
+  copy_file "variants/frontend-stimulus/hello_controller.js", "app/frontend/javascript/controllers/hello_controller.js"
+end
+
+prepend_to_file "app/frontend/packs/application.js" do
+  <<~EO_JS_IMPORTS
+    import { Application } from '@hotwired/stimulus';
+    import { definitionsFromContext } from '@hotwired/stimulus-webpack-helpers';
+  EO_JS_IMPORTS
+end
+
+if TEMPLATE_CONFIG.use_typescript?
+  append_to_file "app/frontend/packs/application.js" do
+    <<~EO_TS_SETUP
+      // TODO
+
+      // Set up stimulus.js https://stimulus.hotwired.dev/
+      // const application = Application.start();
+      // const controllerFilesRegex = /\.js$/u;
+      // const context = require.context('./controllers', true, controllerFilesRegex);
+
+      // application.load(definitionsFromContext(context));
+
+      // // Configure stimulus development experience
+      // application.debug = false;
+      // // window.Stimulus = application;
+    EO_TS_SETUP
+  end
+else
+  append_to_file "app/frontend/packs/application.js" do
+    <<~EO_JS_SETUP
+
+      // Set up stimulus.js https://stimulus.hotwired.dev/
+      const application = Application.start();
+      const controllerFilesRegex = /\.js$/u;
+      const context = require.context('./controllers', true, controllerFilesRegex);
+
+      application.load(definitionsFromContext(context));
+
+      // Configure stimulus development experience
+      application.debug = false;
+      // window.Stimulus = application;
+    EO_JS_SETUP
+  end
+end
+
+# Webpacker does not like const context = require.context('./controllers', true, controllerFilesRegex) - it throws this warning
```

Comment:
I"m doing what https://stimulus.hotwired.dev/handbook/installing#using-webpack-helpers suggests here.

https://github.com/hotwired/stimulus-rails#with-javascript-bundler suggests a different, more explicit approach. Normally I'd be into that but something about having to have boilerplate to load every controller feels un-railsy to me e.g.

```
import HelloController from "./hello_controller"
application.register("hello", HelloController)
```

I'd like to discuss this trade-off and decide what we want here.

Reply:
The problem is that you're passing in a variable, not a RegEx literal, which is not supported because of complexity reasons. Also your controllers need to be placed in `app/frontend/packs/controllers`.

> load every controller feels un-railsy

Understandable, but remember that Rails controllers stay on the backend whereas frontend controllers need to be sent to each client as part of the overall bundle and Webpack cannot know what controllers are actually needed so this will result in every controller being included in every pack.

Reply:
> The problem is that you're passing in a variable, not a RegEx literal Also your controllers need to be placed in app/frontend/packs/controllers

Ah cool, thanks. That was my editor doing some autoformat goodness for me. Fixed now, as is the path to the controllers


> frontend controllers need to be sent to each client as part of the overall bundle and Webpack cannot know what controllers are actually needed so this will result in every controller being included in every pack

Good point. I guess this choice hinges on whether we want less boilerplate or an easier jump to having multiple packs 🤔 . I think we should probably include code for both approaches but comment out the one we don't want as default.

Reply:
I think to start we probably want to go with the less-boilerplate because afaik I don't think we have multiple packs very often and it should be pretty easy to catch (statically, it's literally the count of the number of files in `app/frontend/packs`).